slackbot/bot.py

```diff
@@ -23,13 +31,15 @@ def __init__(self):
             bot_emoji=settings.BOT_EMOJI if hasattr(settings, 'BOT_EMOJI') else None
         )
         self._plugins = PluginsManager()
-        self._dispatcher = MessageDispatcher(self._client, self._plugins)
+        self._dispatcher = Dispatcher(self._client, self._plugins)
 
     def run(self):
         self._plugins.init_plugins()
         self._dispatcher.start()
         self._client.rtm_connect()
         _thread.start_new_thread(self._keepactive, tuple())
+        if py_schedule:
+            _thread.start_new_thread(self._scheduler, tuple())
```

Comment:
Hello. Threads are bad no matter how good you are in thread programming. Take a look at gevent (for py2.7) or take a look at coroutines and uvloop for Python 3.4+

GIL wouldn't given you an ability to run concurrent threads, and threadlock will block main thread completely, so as regular thread does. So, no matter what to use.